sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-notification/e-cert-notification.service.ts

```diff
@@ -0,0 +1,145 @@
+import { Notification, NotificationMessageType, Student } from "@sims/sims-db";
+import { dateDifference } from "@sims/utilities";
+import { EntityManager } from "typeorm";
+
+interface NotificationData {
+  maxCreatedAt: Date;
+  notificationCount: number;
+}
+
+@Injectable()
+export class ECertNotificationService {
+  constructor(
+    private readonly notificationActionsService: NotificationActionsService,
+  ) {}
+
+  /**
+   * Checks and creates disbursement blocked notification.
+   * @param disbursementId disbursement id.
+   * @param studentId student id associated with the blocked disbursement.
+   * @param applicationNumber application number associated with the blocked disbursement.
+   * @param entityManager entity manager to execute in transaction.
+   */
+  async createDisbursementBlockedNotification(
+    disbursementId: number,
+    studentId: number,
+    applicationNumber: string,
+    entityManager: EntityManager,
+  ): Promise<void> {
+    const shouldCreateNotification =
+      await this.shouldCreateDisbursementBlockedNotification(
+        disbursementId,
+        entityManager,
+      );
+    if (shouldCreateNotification) {
+      await this.createDisbursementBlockedNotifications(
+        disbursementId,
+        studentId,
+        applicationNumber,
+        entityManager,
+      );
+    }
+  }
+
+  /**
+   * Checks whether the disbursement blocked notification should be created or not.
+   * @param disbursementId disbursement id.
+   * @param entityManager entity manager to execute in transaction.
+   * @returns boolean indicating if the disbursement blocked notification should be created or not.
+   */
+  private async shouldCreateDisbursementBlockedNotification(
+    disbursementId: number,
+    entityManager: EntityManager,
+  ): Promise<boolean> {
+    const notificationMessageId =
+      NotificationMessageType.StudentNotificationDisbursementBlocked;
+    const notification = await entityManager
+      .getRepository(Notification)
+      .createQueryBuilder("notification")
+      .select("count(*)::int", "notificationCount")
+      .addSelect("max(notification.createdAt)", "maxCreatedAt")
+      .innerJoin("notification.notificationMessage", "notificationMessage")
+      .where("notificationMessage.id = :notificationMessageId", {
+        notificationMessageId,
+      })
+      .andWhere("notification.permanentFailureError IS NULL")
+      .andWhere("notification.metadata->>'disbursementId' = :disbursementId", {
+        disbursementId,
+      })
+      .getRawOne<NotificationData>();
+    const canSendNewNotification =
+      dateDifference(new Date(), notification.maxCreatedAt) >=
+      BLOCKED_DISBURSEMENT_NOTIFICATION_MIN_DAYS_INTERVAL;
+    // Condition check to create notifications: Less than 3 notifications are created previously and there are no failures in sending those created notifications.
+    // Plus, it has been at least 7 days from the last created notification for this disbursement.
```

Comment:
Now  that we have the constants we should remove those hardcoded numbers from the comments.